doc/man3/EVP_PKEY_derive.pod

```diff
@@ -2,20 +2,33 @@
 
 =head1 NAME
 
-EVP_PKEY_derive_init, EVP_PKEY_derive_set_peer, EVP_PKEY_derive - derive public key algorithm shared secret
+EVP_PKEY_derive_init, EVP_PKEY_derive_init_ex, EVP_PKEY_derive_set_peer,
+EVP_PKEY_derive - derive public key algorithm shared secret
 
 =head1 SYNOPSIS
 
  #include <openssl/evp.h>
 
+ int EVP_PKEY_derive_init_ex(EVP_PKEY_CTX *ctx, EVP_EXCHANGE *exchange);
  int EVP_PKEY_derive_init(EVP_PKEY_CTX *ctx);
  int EVP_PKEY_derive_set_peer(EVP_PKEY_CTX *ctx, EVP_PKEY *peer);
  int EVP_PKEY_derive(EVP_PKEY_CTX *ctx, unsigned char *key, size_t *keylen);
 
 =head1 DESCRIPTION
 
-The EVP_PKEY_derive_init() function initializes a public key algorithm
-context using key B<pkey> for shared secret derivation.
+The EVP_PKEY_derive_init_ex() function initializes a public key algorithm
+context for shared secret derivation using the key exchange algorithm
+B<exchange>.
```

Comment:
Could this be just key B<exchange> algorithm.
Maybe Keyexchange is a better name for the type???

Reply:
I find it more confusing to just write key B\<exchange\> algorithm.

I'm open to a different type name, but `EVP_KEYEXCHANGE` seems a little long. Other possibilities could be `EVP_KEYEXCH`, `EVP_KEYEX`, `EVP_KEXCHANGE`

Reply:
I dont like EVP_KEYEX, either of the other 2 are fine.